### PR TITLE
Add GCP to terraform deployment examples

### PR DESCRIPTION
**Pull request by allada**
Adds Google Cloud deployment example to the "deployment-examples" terraform directory.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433)

### CODE REVIEW

**Review by allada — COMMENTED** [review]
+@adam-singer 
+@MarcusSorealheis 

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433#-:-NkYTSkO93LLVNhgmT7y:be0axp9)* status: 0 of 37 files reviewed, all discussions resolved (waiting on @adam-singer and @MarcusSorealheis)

**Review by allada — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433)* status: 0 of 37 files reviewed, 1 unresolved discussion (waiting on @adam-singer and @MarcusSorealheis)

___
*[`deployment-examples/terraform/GCP/README.md` line 3 at r1](https://reviewable.io/reviews/TraceMachina/native-link/433#-NkYTY-lDy1IqRhIFWI_:-NkYTY-lDy1IqRhIFWIa:bpbjoj7) ([raw file](https://github.com/TraceMachina/native-link/blob/c015c06ac48b78f127e9b9b3d072df7ffdc00e57/deployment-examples/terraform/GCP/README.md#L3)):*
> ```Markdown
> # TODO
> 
> Documentation coming soon.
> ```

I don't want to hold this PR up on documentation, since we have other work to do, so going to put the documentation in the backlog.

File: deployment-examples/terraform/GCP/module/instance_cas.tf [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+resource "google_compute_region_instance_template" "cas_instance_template" {
+  name = "${var.project_prefix}-cas-instance-template"
+
+  # Use a very small instance type for the CAS, since it's just a proxy to S3.
+  machine_type   = "e2-highcpu-8"
+  can_ip_forward = false
+
+  service_account {
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
```
Comment (MarcusSorealheis) [inline deployment-examples/terraform/GCP/module/instance_cas.tf]: it's important. No human identity should be tied to permissions in a script.

**Review by allada — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433)* status: 0 of 37 files reviewed, 2 unresolved discussions (waiting on @adam-singer and @MarcusSorealheis)

___
*[`deployment-examples/terraform/GCP/module/instance_cas.tf` line 73 at r2](https://reviewable.io/reviews/TraceMachina/native-link/433#gh-1411796466:-Nka2Vgg1ltq0u_YID8g:b-896fix) ([raw file](https://github.com/TraceMachina/native-link/blob/ffd659d99a92939c6561d8aa7199880f65b58be5/deployment-examples/terraform/GCP/module/instance_cas.tf#L73)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

it's important. No human identity should be tied to permissions in a script.
</blockquote></details>

Done.

**Review by allada — COMMENTED** [review]
Yeah, I would like to just get it in. I have been testing extensively and every time I find a minor issue another issue raises. I won't land it before I'm confident it fully works as-is.

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433#-:-Nka2WlY593vn96w7MYv:b-gdb3oo)* status: 0 of 37 files reviewed, 2 unresolved discussions (waiting on @adam-singer and @MarcusSorealheis)

**Review by adam-singer — APPROVED** [review]
lgtm. Post landing would be great for us all to independently test on a gcp sandbox setup if possible. I do wonder if that will be a bit of churn with the cert init. 

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433#-:-NkbABWDAcPKdGKA7xXg:b-4x013x)* status: 0 of 37 files reviewed, 2 unresolved discussions (waiting on @MarcusSorealheis)

**Review by allada — COMMENTED** [review]
> Dismissed @MarcusSorealheis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433)* status: 0 of 37 files reviewed, all discussions resolved (waiting on @adam-singer and @MarcusSorealheis)

**Review by adam-singer — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433#-:-NkbozvqB87B_xiJm5VC:bnfp4nl)* status: 0 of 37 files reviewed, all discussions resolved (waiting on @MarcusSorealheis)

**Review by allada — COMMENTED** [review]
> Reviewed 29 of 37 files at r1, 2 of 2 files at r2, 3 of 5 files at r3, 1 of 1 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/native-link/433)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @allada)

**Pull request merged**